diff --git a/BUILD b/BUILD
new file mode 100644
index 0000000..390f535
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,17 @@
+
+cc_library(
+    name = "thread_pool",
+    hdrs = ["include/BS_thread_pool.hpp"],
+    strip_include_prefix = "include",
+    includes = ["include"],
+    visibility = ["//visibility:public"],
+)
+
+cc_test(
+    name = "test_thread_pool",
+    srcs = ["tests/BS_thread_pool_test.cpp"],
+    deps = [":thread_pool"],
+    visibility = ["//visibility:public"],
+    env = { "NO_COLOR": "1" },
+    size = "small",
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..6f903c3
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,5 @@
+
+module(
+    name = "bshoshany_thread-pool",
+    version = "5.1.0",
+)
